presidio-anonymizer/presidio_anonymizer/batch_anonymizer_engine.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Dict, Union, Iterable
+import collections
+
+from presidio_anonymizer.entities import DictRecognizerResult
+from presidio_anonymizer import AnonymizerEngine
+from presidio_anonymizer.entities import EngineResult, RecognizerResult
+
+
+class BatchAnonymizerEngine(AnonymizerEngine):
+    """
+    BatchAnonymizerEngine class.
+
+    Class inheriting from the AnonymizerEngine, with additional functionality
+    for anonymizing lists or dictionaries.
+    """
+
+    def anonymize_list(
+        self,
+        texts: List[Union[str, bool, int, float]],
+        recognizer_results_list: List[List[RecognizerResult]],
+        **kwargs
+    ) -> List[EngineResult]:
+        """
+        Anonymize a list of strings.
+
+        :param texts: List containing the texts to be anonymized (original texts)
+        :param recognizer_results_list: A list of lists of RecognizerResult,
+        the output of the AnalyzerEngine on each text in the list.
+        :param kwargs: Additional kwargs for the `AnonymizerEngine.anonymize` method
+        """
+        return_list = []
+        if not recognizer_results_list:
+            recognizer_results_list = [[] for _ in range(len(texts))]
+        for text, recognizer_results in zip(texts, recognizer_results_list):
+            if type(text) in (str, bool, int, float):

```

Comment:
When is it not str?

Reply:
Oh, if used by the anonymize_dict? 
In that case, how would float and bool be PIIs? why sending them for anonymization?

Reply:
int I get, but how is float when its not in string representation?

Reply:
It could be float in the database, but it's actually someone's age for example.

Reply:
Right :)